pkg/controller/managedresources/health/health_controller.go

```diff
@@ -72,18 +72,35 @@ func (r *HealthReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	resourcesObjectReferences := mr.Status.Resources
 	for _, ref := range resourcesObjectReferences {
-		obj := &unstructured.Unstructured{}
-		obj.SetAPIVersion(ref.APIVersion)
-		obj.SetKind(ref.Kind)
+		// only retrieve object if we will be able to check its health
```

Comment:
Hm, can't we at least check if the object exists instead of silently continuing?

Reply:
Good catch, fixed, PTAL :)